support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListContainer.tsx

```diff
@@ -41,21 +46,25 @@ export function CheckoutBenefitsListContainer({
 	const dispatch = useContributionsDispatch();
 
 	const contributionType = useContributionsSelector(getContributionType);
-	if (isOneOff(contributionType)) {
-		return null;
-	}
 
-	const { countryGroupId, currencyId } = useContributionsSelector(
+	const { countryGroupId, countryId, currencyId } = useContributionsSelector(
 		(state) => state.common.internationalisation,
 	);
+	const abParticipations = initAbTests(
+		countryId,
+		countryGroupId,
+		getSettings(),
+	);
 	const selectedAmount = useContributionsSelector(getUserSelectedAmount);
 	const minimumContributionAmount = useContributionsSelector(
 		getMinimumContributionAmount,
 	);
 
 	const currency = currencies[currencyId];
 
-	const thresholdPrice = getThresholdPrice(countryGroupId, contributionType);
+	const thresholdPrice = isRecurring(contributionType)
+		? getThresholdPrice(countryGroupId, contributionType)
+		: 0;
```

Comment:
When do we want the thresholdPrice at 0?

Reply:
Its used for the handleButtonClick (event handler line86), it need's a valid threshold price even when the contributionType is ONE_OFF, otherwise I get  'possibly undefined' in handleButtonClick. Can look into a better way however tomorrow.

Reply:
Did you get a typescript error prompting you to make the change? This code would have never actually run when the `contributionType` was ONE_OFF because of the return statement `if (isOneOff(contributionType)) { return null }` that's been removed from this module. I'd be very wary of this change. I don't think we should be repurposing the `checkoutBenefitsListContainer.tsx` component to also double up as the nudge component. I'd create a brand new component for the nudge component that only renders when `isOneOff(contributionType)` and the user is in the appropriate AB test variant.

Reply:
Yes, agree this would be much neater and may help or/simplfy the display side effects have seen, will bring up the nudge component out of the benefits list.

Reply:
Ok, finished completing this, would you mind having a look it's acceptable? Have asked Maria to look too. Doing this completely got rid of the display side-effects so will remember components go as high as they can in future.

Reply:
Completed